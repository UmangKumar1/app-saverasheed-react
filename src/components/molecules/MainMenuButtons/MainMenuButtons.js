import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes, css, cx } from 'react-emotion'
import Link from '../../atoms/Link'

const { bool, shape, number } = PropTypes

const ButtonWrapper = styled('div')({
  marginBottom: 50,
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
})

const topButtonAnim = keyframes`
0% {
  transform: translateX(-1000%);
}

100% {
  transform: translateX(0%);
}
`

const bottomButtonAnim = keyframes`
0% {
  transform: translateX(1000%);
}

100% {
  transform: translateX(0%);
}
`

const buttonAnimInstant = keyframes`
0%, 100% {
  transform: translateX(0%);
}
`

const animationTimingFunction = 'cubic-bezier(0.22, 0.61, 0.36, 1)'

const createAnimCx = ({ duration, delay }) =>
  css({
    transform: 'translateX(-1000%)',
    animationDuration: `${duration}ms`,
    animationDelay: `${delay}ms`,
    animationFillMode: 'forwards',
    animationTimingFunction,
  })

const topAnimCx = css({
  animationName: topButtonAnim,
})

const bottomAnimCx = css({
  animationName: bottomButtonAnim,
})

const skipCx = css({ animation: `${buttonAnimInstant} 1ms forwards` })

const MenuButton = styled('button')({
  color: 'inherit',
  border: 'none',
  backgroundColor: 'transparent',
  fontSize: 32,
  margin: 10,
})

const MainMenuButtons = ({ skip, topAnim, bottomAnim }) => (
  <ButtonWrapper>
    <Link
      to="/main/signin"
      className={cx(createAnimCx(topAnim), topAnimCx, { [skipCx]: skip })}
    >
      <MenuButton>Returning Player</MenuButton>
    </Link>
    <Link
      to="/main/signup"
      className={cx(createAnimCx(bottomAnim), bottomAnimCx, { [skipCx]: skip })}
    >
      <MenuButton>New Player</MenuButton>
    </Link>
  </ButtonWrapper>
)

const anim = shape({ duration: number, delay: number })

MainMenuButtons.propTypes = {
  skip: bool,
  topAnim: anim.isRequired,
  bottomAnim: anim.isRequired,
}

export default MainMenuButtons
