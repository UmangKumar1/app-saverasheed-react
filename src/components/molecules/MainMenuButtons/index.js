export { default, default as MainMenuButtons } from './MainMenuButtons'
