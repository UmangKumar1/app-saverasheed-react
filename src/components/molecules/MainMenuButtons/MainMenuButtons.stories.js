import React from 'react'
import { storiesOf } from '@storybook/react'
import MainMenuButtons from '.'

const topButtonAnimDelay = 0
const topButtonAnimDuration = 1000

const bottomButtonAnimDelay = topButtonAnimDelay + topButtonAnimDuration
const bottomButtonAnimDuration = 1000

const topAnim = {
  duration: topButtonAnimDuration,
  delay: topButtonAnimDelay,
}

const bottomAnim = {
  duration: bottomButtonAnimDuration,
  delay: bottomButtonAnimDelay,
}

const props = { topAnim, bottomAnim }

storiesOf('MainMenuButtons', module)
  .add('with anim', () => <MainMenuButtons {...props} />)
  .add('skip anim', () => <MainMenuButtons skip {...props} />)
