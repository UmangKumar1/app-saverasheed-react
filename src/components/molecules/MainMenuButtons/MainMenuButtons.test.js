import React from 'react'
import { shallow } from 'enzyme'

import MainMenuButtons from '.'

const wrap = (props = {}) => shallow(<MainMenuButtons {...props} />)

const topButtonAnimDelay = 0
const topButtonAnimDuration = 1000

const bottomButtonAnimDelay = topButtonAnimDelay + topButtonAnimDuration
const bottomButtonAnimDuration = 1000

const topAnim = {
  duration: topButtonAnimDuration,
  delay: topButtonAnimDelay,
}

const bottomAnim = {
  duration: bottomButtonAnimDuration,
  delay: bottomButtonAnimDelay,
}

const props = { topAnim, bottomAnim }

describe('MainMenuButtons', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap(props)
    expect(wrapper).toMatchSnapshot()
  })
})
