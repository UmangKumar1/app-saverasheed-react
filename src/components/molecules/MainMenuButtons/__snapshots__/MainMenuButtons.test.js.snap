// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuButtons renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainMenuButtons
    bottomAnim={
      Object {
        "delay": 1000,
        "duration": 1000,
      }
    }
    topAnim={
      Object {
        "delay": 0,
        "duration": 1000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Link)
          className="css-ln2ph0"
          replace={false}
          to="/main/signin"
        >
          <Styled(button)>
            Returning Player
          </Styled(button)>
        </Styled(Link)>,
        <Styled(Link)
          className="css-14sjvqw"
          replace={false}
          to="/main/signup"
        >
          <Styled(button)>
            New Player
          </Styled(button)>
        </Styled(Link)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(button)>
            Returning Player
          </Styled(button)>,
          "className": "css-ln2ph0",
          "replace": false,
          "to": "/main/signin",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Returning Player",
          },
          "ref": null,
          "rendered": "Returning Player",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(button)>
            New Player
          </Styled(button)>,
          "className": "css-14sjvqw",
          "replace": false,
          "to": "/main/signup",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "New Player",
          },
          "ref": null,
          "rendered": "New Player",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Link)
            className="css-ln2ph0"
            replace={false}
            to="/main/signin"
          >
            <Styled(button)>
              Returning Player
            </Styled(button)>
          </Styled(Link)>,
          <Styled(Link)
            className="css-14sjvqw"
            replace={false}
            to="/main/signup"
          >
            <Styled(button)>
              New Player
            </Styled(button)>
          </Styled(Link)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(button)>
              Returning Player
            </Styled(button)>,
            "className": "css-ln2ph0",
            "replace": false,
            "to": "/main/signin",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Returning Player",
            },
            "ref": null,
            "rendered": "Returning Player",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(button)>
              New Player
            </Styled(button)>,
            "className": "css-14sjvqw",
            "replace": false,
            "to": "/main/signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "New Player",
            },
            "ref": null,
            "rendered": "New Player",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
